llvm/include/llvm/IR/InlineAsm.h

```diff
@@ -528,6 +528,20 @@ class InlineAsm final : public Value {
   }
 };
 
+/// Inline Asm specifies input & output constraint which can
+/// specifiy target specific criteria for operand. If this criteria
+/// does not match, we must throw error.
+/// NO_ERROR represents Operand constraints are valid/applicable
+/// OPERAND_ERROR represents some constraint(unspecified) failed
+/// UNKNOWN_MODIFIER_ERROR represents use of unknown char constraint
+/// CONSTRAINT_<char>_ERROR represents error regarding constraint.
+enum class AsmOperandErrorCode {
+  NO_ERROR = 0,
+  OPERAND_ERROR,
+  UNKNOWN_MODIFIER_ERROR,
+  CONSTRAINT_H_ERROR,
```

Comment:
Why can't we just make the constraint code available to the error printing code? Having one of these for every possible case sounds like a headache.

Reply:
All constraint codes related to inline asm are defined in same file. 
Will adding `AsmPrinter.h`   reasonable ?

Reply:
The constraint H seems too specific. Why not just let this be an arbitrary string carried by Error?